### ISSUE TITLE
NOTE_018 Interface cleanup

### ISSUE THREAD

**Issue by deveah** [sw20-tug/SWT-Morning-3#84]
The latest added features do not have a proper interface.

New interface elements are required for the features. Accessibility should be taken into account.
